lib/kcl/workers/consumer.rb

```diff
@@ -24,13 +25,15 @@ def consume!
 
       loop do
         result = safe_get_records(shard_iterator)
-
-        records_input = create_records_input(
-          result[:records],
-          result[:millis_behind_latest],
-          record_checkpointer
-        )
-        @record_processor.process_records(records_input)
+        if result != @last_result # only process results if they're different
```

Comment:
need to verify this comparison is valid (and we don't mistake metadata for content)

maybe result[:records] == @old_result[:records] is right